### ISSUE TITLE
Antivirus software reporting

### ISSUE THREAD

**Issue by monkeycc** [gradio-app/gradio#3230]
![微信截图_20230218104922](https://user-images.githubusercontent.com/6490927/219827938-f963ec60-7591-4c1b-b43d-2c608f7245c7.png)

`gradio\frpc_windows_amd64`

https://weishi.360.cn/jisu/

https://www.360totalsecurity.com/

**Comment by abidlabs** [comment]
Hmm thanks for reporting. I'm not sure if there's anything we can do here except to confirm that `frpc_windows_amd64` is not a virus, it's needed to set `share=True`.

**Comment by ecker00** [comment]
Also got this reported, and it's flagged by 36 of 68 vendors on VirusTotal: https://www.virustotal.com/gui/file/c9db0db0f0f41ce3fe84f92a785d4ba0ab351ee231ce9e53924a168889a525fd

What exactly is this file, and why would it be flagged?

**Comment by abidlabs** [comment]
Hi @ecker00 it's a file that allows you to create a public link for your Gradio app (when you set `share=True` in your `launch()` method)

**Comment by erhanX** [comment]
@abidlabs It is also detected by Defender and quarantined so it is not really helping on Windows unless you give instruction how to whitelist the file or get it out of the virus DBs or just don't care. ;)

**Comment by abidlabs** [comment]
@erhanX can you try these steps: https://www.makeuseof.com/how-to-whitelist-files-windows-defender/?

**Comment by erhanX** [comment]
I know how to whitelist. It is not about me. I am just notifying you that any normal Windows user will have this problem. Maybe it was not detected before by Defender when the other person reported it. 

Maybe make it an optional package which first gets installed when public=True is used first time and put this information there. Or leave it as it is because people who dont use it will not notice anyway.

**Comment by abidlabs** [comment]
> Maybe make it an optional package which first gets installed when public=True is used first time

That is the current behavior actually

> and put this information there

That's a good idea, we'll do that, thanks!

**Comment by robertbozsa** [comment]
Avast also sees this file as malware and moves it to quarantine.

File: frpc_windows_amd64_v0.2
Threats: FileRepMalware & Win64:Malware-gen
Path: C:\Users\user1\AppData\Local\Programs\Python\Python310\Lib\site-packages\gradio\frpc_windows_amd64_v0.2

Although it is moved to quarantine the application still seems to work. Is this file necessary for something?

This was installed as part of the process to setup a custom knowledgebase for ChatGPT.

https://beebom.com/how-train-ai-chatbot-custom-knowledge-base-chatgpt-api/

**Comment by gitihobo** [comment]
Multiple anti-virus detect as riskware, anything that can be done about that? How about another script that removes any unnecessary components if you wish to make your environment completely offline?

**Comment by ayoubachak** [comment]
> > Maybe make it an optional package which first gets installed when public=True is used first time
> 
> That is the current behavior actually
> 
> > and put this information there
> 
> That's a good idea, we'll do that, thanks!

Can you please add a warning for corporate computers, the file singlehandedly blocked every activity in my computer ( antivirus detected it ), now it's taken for maintenance.
I'm pretty sure I won't be the only one who had this issue, I have to mention that I myself went and downloaded the file and it wasn't detected as a malware in the beginning.

**Comment by carlthome** [comment]
The enterprise security department where I work also alerted on this.

**Comment by niccokingdom** [comment]
Same here, flagged by corporate, isn't it possible to fix the file, that seems like one of the main straightforward features to be able to share apps? Thanks

**Comment by salvasisua2** [comment]
![image](https://github.com/gradio-app/gradio/assets/149637952/c5f28ff1-751b-428c-8e36-38db8baeb940)


McAfee detect...

**Comment by Woahai321** [comment]
SentinelOne also reports this as malware.

**Comment by danmack** [comment]
crowdstrike falcon also quarantines this as Adware/Malware.

**Comment by mulder1** [comment]
Frp component is a reverse proxy written by a Chinese programmer.
It's like saying welcome to trouble if you install it on your computer. No sane corporate environment will allow that

**Comment by nucklearproject** [comment]
Same problem here 😴 
![image](https://github.com/user-attachments/assets/5cadee0e-d56b-4819-b796-b670eefa29c1)

**Comment by jblouin** [comment]
Symantec also reports this as a security issue
![symantec_gradio_frpc_issue](https://github.com/user-attachments/assets/62341064-b947-4ba0-9451-8c5b1df24f05)

**Comment by cartolid** [comment]
Cytomic EPDR also reports it as PUP/generic (and deletes it).

**Comment by FurkanGozukara** [comment]
my kaspersky antivirus reporting as like this perhaps they can be  modified to not report

not super important but just reporting

`Today, 29/09/2024 13:11:53;G:\Kosmos-2\venv\Lib\site-packages\gradio\frpc_windows_amd64_v0.2;Not processed;Disinfection not possible;not-a-virus:HEUR:NetTool.Win64.FRP.gen;Postponed;File;G:\Kosmos-2\venv\Lib\site-packages\gradio;frpc_windows_amd64_v0.2;Not processed;Legitimate software that can be used by intruders to damage your computer or personal data;Low;Partially;SECoursesPC\Furkan;Active user`

**Comment by SH4DY** [comment]
Also just ran into this on my corporate machine. Crowdstrike on Apple Silicon.

**Comment by rsxdalv** [comment]
I don't think this issue should be closed since antiviruses still report it. I'd also prefer if frpc was an optional module so that users can install it, rather than everyone getting a warning.

**Comment by sunilagarwal2007** [comment]
Has anyone experienced any issues with the Gradio library being flagged as potentially malicious? 

While using Gradio on my corporate machine for learning and experimenting with Machine Learning models, it identified a malicious application/tools on system.:

/lib/python3.9/site-packages/gradio/frpc_darwin_arm64_v0.2

I'd appreciate any insights on why this might be occurring, Is Gradio has anything malicious library running at backgroud ?

**Comment by Scott-Kelly** [comment]
I am using the share option w gradio 0.3 version of this file, white listed it, and the public url is still not being published.

**Comment by maxhillaert** [comment]
This is flagged in my org too.  Every large org with an IT Risk department will block this.    This should be optional or another solution chosen.

**Comment by rsxdalv** [comment]
> This is flagged in my org too. It's on 23 threat databases our IT Risk department has said. Every large org with an IT Risk department will block this. This should be optional or another solution chosen.

And it basically should be, I still think another package could use this "good" fprc to do "bad" things. Unless it's somehow locked to only work for one purpose within gradio and be impossible to access otherwise, I doubt anyone is going to be unblocking it anytime soon, let alone how can they tell if this is the "good" or "bad" fprc.
What I can propose to you is to fork gradio and make a version without this file, that would be my corporate go to solution. (Ok it would probably be something convoluted like deleting the file from existing packages). After all, it's not a hard requirement.

**Comment by maxhillaert** [comment]
@rsxdalv  yes, that is the solution  I will pursue.  RIght now I'm just doing Shared=False  which avoids the downloading of this.  Of course as soon as someone decides to turn that flag on, the alarms will blare again, so if I go for gradio , long term I will fork this out

**Comment by akash-gupta-parloa** [comment]
got this issue in SentinealOne on Apple Silicon for file frpc_darwin_arm64_v0.3

**Comment by eliam** [comment]
Kaspersky virus removal tool

<img width="833" alt="Image" src="https://github.com/user-attachments/assets/94683eb8-5786-431e-89f9-4a0a76ffe63d" />

**Comment by Twodragon0** [comment]
Hi, we’re experiencing the same issue with our corporate security—our antivirus flags frpc_darwin_arm64_v0.3 as malware, which disrupts Gradio’s share functionality. Is there an official fix or can this component be made optional? Thanks!

**Comment by abidlabs** [comment]
Hi folks who are experiencing the antivirus issue, can you try installing `gradio` like this:

```py
pip install https://gradio-pypi-previews.s3.amazonaws.com/4814351d6866e31cdff70ebd87823e6fa10dcf2b/gradio-5.23.3-py3-none-any.whl
```

and see if it resolves the issue for you?

**Comment by monkeycc** [comment]
`pip install https://gradio-pypi-previews.s3.amazonaws.com/4814351d6866e31cdff70ebd87823e6fa10dcf2b/gradio-5.23.3-py3-none-any.whl
`

```python
import gradio as gr

def greet(name, intensity):
    return "Hello, " + name + "!" * int(intensity)

demo = gr.Interface(
    fn=greet,
    inputs=["text", "slider"],
    outputs=["text"],
)

demo.launch(share=True)

```

```log
* Running on local URL:  http://127.0.0.1:7860

Could not create share link. Missing file: D:\gradio\frpc\frpc_windows_amd64_v0.3. 

Please check your internet connection. This can happen if your antivirus software blocks the download of this file. You can install manually by following these steps: 

1. Download this file: https://cdn-media.huggingface.co/frpc-gradio-0.3/frpc_windows_amd64.exe
2. Rename the downloaded file to: frpc_windows_amd64_v0.3
3. Move the file to this location: D:\gradio\frpc
```

frpc_windows_amd64.exe
Report poison
@abidlabs 

<img width="345" alt="Image" src="https://github.com/user-attachments/assets/1b12e78d-d2e8-4514-a2bb-5f83dab529e1" />

**Comment by purplemountain** [comment]
Norton 360 reports this is as a malware:

![Image](https://github.com/user-attachments/assets/93426623-8aff-477d-9d99-09c0605f75a0)

**Comment by Berengames** [comment]
https://www.virustotal.com/gui/file/14bc0ea470be5d67d79a07412bd21de8a0a179c6ac1116d7764f68e942dc9ceb

**Comment by JordiCorbilla** [comment]
Same error here, is there a fix?

```
Error starting the application: [WinError 225] Operation did not complete successfully because the file contains a virus or potentially unwanted software: 'C:\\Users\\site-packages\\gradio\\frpc_windows_amd64_v0.2'
```

**Comment by rsxdalv** [comment]
Yes, it's documented in this thread. Please reread it. Abd there's always
the possibility to fork the project, and install your fork, since pip just
wants any gradio that matches the version spec. So you can pip install
git+your-gradio-fork.git

**Comment by JordiCorbilla** [comment]
> Yes, it's documented in this thread. Please reread it. Abd there's always
> the possibility to fork the project, and install your fork, since pip just
> wants any gradio that matches the version spec. So you can pip install
> git+your-gradio-fork.git
> […](#)

thank you! got it working now

**Comment by elismasilva** [comment]
Hey here it is on AVAST <img width="545" height="302" alt="Image" src="https://github.com/user-attachments/assets/c9af9037-2924-44e9-acea-fc7b4b908a9c" />

**Comment by FurkanGozukara** [comment]
ye this is a false positive still happening sometimes 

i am getting user reports. i wish we could prevent this. people getting worried
